packages/api/cms-api/src/common/filter/mikro-orm.ts

```diff
@@ -274,10 +274,15 @@ export function searchToMikroOrmQuery(search: string, fieldsOrMetadata: string[]
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
             const or: any = {};
             let nestedFilter = or;
-            for (const fieldPart of field.split(".")) {
+            const fieldParts = field.split(".");
+            const column = fieldParts.pop();
+            if (column === undefined) {
+                continue;
+            }
+            for (const fieldPart of fieldParts) {
                 nestedFilter = nestedFilter[fieldPart] = {};
             }
-            nestedFilter.$ilike = quotedSearch;
+            nestedFilter[raw((alias) => `${alias}."${column}"::text`)] = { $ilike: quotedSearch };
```

Comment:
Is it okay to always cast to text here? Otherwise I'd need to add a "needs cast" information to the field.

Reply:
Yes, I can access both type and columnType(s).

Reply:
I'd really prefer to avoid unneeded ::text then.